internal/clients/equinix.go

```diff
@@ -113,6 +120,25 @@ func TerraformSetupBuilder(tfProvider *schema.Provider) terraform.SetupFn {
 		}
 
 		ps.Configuration = prepareTerraformProviderConfiguration(equinixCreds, pc.Spec.Configuration)
-		return ps, nil
+		return ps, errors.Wrap(configureNoForkEquinixClient(ctx, &ps, *setupCfg.TerraformProvider), "failed to configure the no-fork equinix client")
 	}
 }
+
+func configureNoForkEquinixClient(ctx context.Context, ps *terraform.Setup, p schema.Provider) error {
```

Comment:
I'm only copying what I see in other providers and I'm not sure why this should be added.

The NoFork concept emerged in https://github.com/crossplane/upjet/pull/294

> Both clients employ an in-memory Terraform state cache as these clients no longer construct Terraform workspaces while reconciling their resources. We have an initial evaluation of the memory cost of this cache with 10k MRs and it's in the order of a few hundred megabytes (~200 MB), much smaller than the footprint of the process fork-based clients.

After the introduction of Framework support in Upjet https://github.com/crossplane/upjet/pull/329, NoFork was renamed to TerraformPluginSDK. https://github.com/crossplane/upjet/pull/338